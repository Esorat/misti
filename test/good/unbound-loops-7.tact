fun test() {
    let i: Int = 0;
    while (i > 0) {} // warning: unbound loop
}
