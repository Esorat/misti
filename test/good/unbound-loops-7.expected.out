test/good/unbound-loops-7.tact:2:9:
  1 | fun test() {
> 2 |     let i: Int = 0;
              ^
  3 |     while (i > 0) {} // warning: unbound loop
Read-only variable
Help: Consider creating a constant instead
See: https://nowarp.github.io/tools/misti/docs/detectors/ReadOnlyVariables

test/good/unbound-loops-7.tact:3:12:
  2 |     let i: Int = 0;
> 3 |     while (i > 0) {} // warning: unbound loop
                 ^
  4 | }
Unbounded Loop: The condition variable doesn't change within the loop
Help: Consider changing the variable within the loop to ensure it terminates
See: https://nowarp.github.io/tools/misti/docs/detectors/UnboundLoops